node/pkg/reporter/utils_test.go

```diff
@@ -1,307 +1,58 @@
-// 			"address":            "0x1234" + strconv.Itoa(i),
-// 			"submit_interval":    TestInterval,
-// 			"fetch_interval":     TestInterval,
-// 			"aggregate_interval": TestInterval})
-// 		if err != nil {
-// 			t.Fatalf("QueryRow failed: %v", err)
-// 		}
-// 		tmpConfigs = append(tmpConfigs, tmpConfig)
-// 	}
-
-// 	aggregates := []GlobalAggregate{
-// 		{ConfigID: tmpConfigs[0].ID, Round: 1},
-// 		{ConfigID: tmpConfigs[1].ID, Round: 2},
-// 		{ConfigID: tmpConfigs[2].ID, Round: 3},
-// 	}
-
-// 	proofMap := map[int32][]byte{
-// 		tmpConfigs[0].ID: []byte("proof1"),
-// 		tmpConfigs[1].ID: []byte("proof2"),
-// 	}
-
-// 	expectedUpsertRows := [][]any{
-// 		{tmpConfigs[0].ID, int32(1), []byte("proof1")},
-// 		{tmpConfigs[1].ID, int32(2), []byte("proof2")},
-// 	}
-// 	err := UpsertProofs(ctx, aggregates, proofMap)
-// 	if err != nil {
-// 		t.Fatalf("UpsertProofs failed: %v", err)
-// 	}
-// 	result, err := db.QueryRows[Proof](ctx, "SELECT config_id, round, proof FROM proofs WHERE config_id IN ("+strconv.Itoa(int(tmpConfigs[0].ID))+", "+strconv.Itoa(int(tmpConfigs[1].ID))+")", nil)
-// 	if err != nil {
-// 		t.Fatalf("QueryRows failed: %v", err)
-// 	}
-
-// 	for i, p := range result {
-// 		assert.Equal(t, expectedUpsertRows[i], []any{p.ConfigID, p.Round, p.Proof})
-// 	}
-
-// 	proofMap = map[int32][]byte{
-// 		tmpConfigs[0].ID: []byte("proof3"),
-// 		tmpConfigs[1].ID: []byte("proof4"),
-// 	}
-// 	expectedUpsertRows = [][]any{
-// 		{tmpConfigs[0].ID, int32(1), []byte("proof3")},
-// 		{tmpConfigs[1].ID, int32(2), []byte("proof4")},
-// 	}
-// 	err = UpdateProofs(ctx, aggregates, proofMap)
-// 	if err != nil {
-// 		t.Fatalf("UpdateProofs failed: %v", err)
-// 	}
-// 	result, err = db.QueryRows[Proof](ctx, "SELECT config_id, round, proof FROM proofs WHERE config_id IN ("+strconv.Itoa(int(tmpConfigs[0].ID))+", "+strconv.Itoa(int(tmpConfigs[1].ID))+")", nil)
-// 	if err != nil {
-// 		t.Fatalf("QueryRows failed: %v", err)
-// 	}
+package reporter
 
-// 	for i, p := range result {
-// 		assert.Equal(t, expectedUpsertRows[i], []any{p.ConfigID, p.Round, p.Proof})
-// 	}
-// }
+import (
+	"reflect"
+	"testing"
+)
+
+func TestProcessDalWsRawData(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    RawSubmissionData
+		expected SubmissionData
+		wantErr  bool
+	}{
+		{
+			name: "Valid input",
+			input: RawSubmissionData{
+				Value:         "123",
+				AggregateTime: "1609459200",
+				Proof:         "0xabcdef",
+				FeedHash:      "0x123456",
+			},
+			expected: SubmissionData{
+				Value:         123,
+				AggregateTime: 1609459200,
+				Proof:         []byte{0xab, 0xcd, 0xef},
+				FeedHash:      [32]byte{0x12, 0x34, 0x56},
+			},
+			wantErr: false,
+		},
+		{
+			name: "Invalid value",
+			input: RawSubmissionData{
+				Value:         "abc",
+				AggregateTime: "1609459200",
+				Proof:         "0xabcdef",
+				FeedHash:      "0x123456",
+			},
+			expected: SubmissionData{},
+			wantErr:  true,
+		},
```

Comment:
can you add few more cases with empty values?

Reply:
done!

Reply:
think you didn't push yet?

Reply:
just did :)